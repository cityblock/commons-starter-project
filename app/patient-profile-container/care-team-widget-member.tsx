import * as classNames from 'classnames';
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import { FullUserFragment } from '../graphql/types';
import * as styles from './css/care-team-widget-member.css';

interface IProps {
  careTeamMember: FullUserFragment;
  selected: boolean;
  onClick: (careTeamMemberId: string) => any;
}

const generateIconStyles = (iconStyle: string): string =>
  classNames(styles.careTeamContactIcon, iconStyle);

const slackIconStyles = generateIconStyles(styles.careTeamSlackIcon);
const phoneIconStyles = generateIconStyles(styles.careTeamPhoneIcon);
const emailIconStyles = generateIconStyles(styles.careTeamEmailIcon);

export default class CareTeamWidgetMember extends React.Component<IProps, {}> {
  formatCareTeamMemberRole(role: string): string {
    return role.charAt(0).toUpperCase() + role.slice(1);
  }

  render() {
    const { careTeamMember, onClick, selected } = this.props;

    const careTeamMemberPhotoUrl = careTeamMember.googleProfileImageUrl || '';
    const careTeamMemberFullName = `${careTeamMember.firstName} ${careTeamMember.lastName}`;
    const careTeamMemberRole = this.formatCareTeamMemberRole(careTeamMember.userRole);
    const careTeamPhoneNumber = careTeamMember.phone;

    const contactStyles = classNames(styles.careTeamMemberContact, {
      [styles.open]: selected,
      [styles.phone]: !!careTeamPhoneNumber,
    });

    return (
      <div className={styles.careTeamMemberRow}>
        <div className={styles.careTeamMemberDetails} onClick={() => onClick(careTeamMember.id)}>
          <div className={styles.careTeamMemberPhoto}>
            <img src={careTeamMemberPhotoUrl} />
          </div>
          <div className={styles.careTeamMemberLabel}>
            <div className={styles.careTeamMemberName}>{careTeamMemberFullName}</div>
            <div className={styles.careTeamMemberRole}>{careTeamMemberRole}</div>
          </div>
        </div>
        <div className={contactStyles}>
          <div className={styles.careTeamMemberContactRow}>
            <div className={slackIconStyles} />
            <FormattedMessage id="careTeam.slack">
              {(message: string) => <div className={styles.careTeamContactLabel}>{message}</div>}
            </FormattedMessage>
          </div>
          {careTeamPhoneNumber && (
            <div className={styles.careTeamMemberContactRow}>
              <div className={phoneIconStyles} />
              <div className={styles.careTeamContactLabel}>{careTeamPhoneNumber}</div>
            </div>
          )}
          <div className={styles.careTeamMemberContactRow}>
            <div className={emailIconStyles} />
            <FormattedMessage id="careTeam.email">
              {(message: string) => <div className={styles.careTeamContactLabel}>{message}</div>}
            </FormattedMessage>
          </div>
        </div>
      </div>
    );
  }
}
