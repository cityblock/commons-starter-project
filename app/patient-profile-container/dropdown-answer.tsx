import * as classNames from 'classnames';
import * as React from 'react';
import { FullAnswerFragment, FullQuestionFragment } from '../graphql/types';
import * as formStyles from '../shared/css/forms.css';
import * as styles from './css/risk-areas.css';

interface IProps {
  currentAnswer: { id: string; value: string };
  question: FullQuestionFragment;
  onChange: (questionId: string, answerId: string, value: string | number) => any;
}

export default class DropdownAnswer extends React.Component<IProps, {}> {
  constructor(props: IProps) {
    super(props);

    this.renderDropdownOption = this.renderDropdownOption.bind(this);
    this.onDropdownChange = this.onDropdownChange.bind(this);
  }

  renderDropdownOption(answer: FullAnswerFragment, index: number) {
    return (
      <option key={`${answer.id}-${index}`} value={answer.value}>
        {answer.displayValue}
      </option>
    );
  }

  onDropdownChange(value: string | number) {
    const { question, onChange } = this.props;

    const chosenAnswer = (question.answers || []).find(answer => answer.value === value);

    if (chosenAnswer) {
      onChange(question.id, chosenAnswer.id, value);
    }
  }

  render() {
    const { question, currentAnswer } = this.props;
    const answers = question.answers || [];

    return (
      <div className={styles.riskAssessmentQuestionBody}>
        <select
          value={currentAnswer ? currentAnswer.value : 'Select one'}
          onChange={event => this.onDropdownChange(event.currentTarget.value)}
          className={classNames(formStyles.select, styles.select)}
        >
          <option value={'Select one'} disabled={true}>
            Select one
          </option>
          {answers.map(this.renderDropdownOption)}
        </select>
      </div>
    );
  }
}
