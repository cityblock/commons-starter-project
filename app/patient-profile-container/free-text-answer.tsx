import * as React from 'react';
import { FullQuestionFragment } from '../graphql/types';
import * as formStyles from '../shared/css/forms.css';
import * as styles from './css/risk-areas.css';

interface IProps {
  currentAnswer: { id: string; value: string; };
  question: FullQuestionFragment;
  onChange: (questionId: string, answerId: string, value: string | number) => any;
}

export default class FreeTextAnswer extends React.Component<IProps, {}> {
  render() {
    const { question, currentAnswer, onChange} = this.props;
    const answers = question.answers || [];

    // This is a weird answer type. Return nothing if there are no answers associated with it.
    const answer = answers[0];

    if (answers) {
      const answerId = answer.id;

      return (
        <div className={styles.riskAssessmentQuestionBody}>
          <div className={styles.textArea}>
            <textarea
              value={currentAnswer ? currentAnswer.value : ''}
              onChange={event => onChange(question.id, answerId, event.target.value)}
              className={formStyles.textarea} />
          </div>
        </div>
      );
    } else {
      return null;
    }
  }
}
