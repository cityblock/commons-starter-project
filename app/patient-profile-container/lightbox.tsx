import * as classNames from 'classnames';
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import * as styles from './css/lightbox.css';

interface IProps {
  images: string[];
  isOpen: boolean;
  openingImage?: string;
  onDismiss: () => any;
}

interface IState {
  selectedImageIndex: number;
  imagesLength?: number;
}

export default class Lightbox extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);

    this.onBackClick = this.onBackClick.bind(this);
    this.onNextClick = this.onNextClick.bind(this);
    this.onKeyDown = this.onKeyDown.bind(this);

    const { openingImage, images } = props;

    const selectedImageIndex = this.getOpeningImageIndex(openingImage, images);

    this.state = { selectedImageIndex };
  }

  componentWillUnmount() {
    // TODO: maybe overkill?
    document.removeEventListener('keydown', this.onKeyDown as any);
  }

  componentWillReceiveProps(nextProps: IProps) {
    const { images, isOpen, openingImage } = nextProps;

    if (isOpen) {
      document.addEventListener('keydown', this.onKeyDown as any);

      const selectedImageIndex = this.getOpeningImageIndex(openingImage, images);
      this.setState({ imagesLength: images.length, selectedImageIndex });
    } else {
      document.removeEventListener('keydown', this.onKeyDown as any);
    }
  }

  getOpeningImageIndex(openingImage: string | undefined, imagesList: string[]) {
    let foundIndex: number = 0;

    if (openingImage) {
      foundIndex = imagesList.indexOf(openingImage);
    }

    return foundIndex === -1 ? 0 : foundIndex;
  }

  onBackClick() {
    const { selectedImageIndex } = this.state;

    if (selectedImageIndex > 0) {
      this.setState({ selectedImageIndex: selectedImageIndex - 1 });
    }
  }

  onNextClick() {
    const { selectedImageIndex } = this.state;
    const { images } = this.props;
    const imagesLength = images.length;

    if (selectedImageIndex < imagesLength - 1) {
      this.setState({ selectedImageIndex: selectedImageIndex + 1 });
    }
  }

  onKeyDown(event: React.KeyboardEvent<KeyboardEvent>) {
    const { onDismiss } = this.props;

    if (event.keyCode === 27) {
      // Esc pressed
      onDismiss();
    } else if (event.keyCode === 37) {
      // Left arrow pressed
      this.onBackClick();
    } else if (event.keyCode === 39) {
      // Right arrow pressed
      this.onNextClick();
    }
  }

  render() {
    const { selectedImageIndex } = this.state;
    const { isOpen, images, onDismiss } = this.props;
    const imagesLength = images.length;

    const backArrowStyles = classNames(styles.backArrow, {
      [styles.hidden]: selectedImageIndex === 0,
    });

    const nextArrowStyles = classNames(styles.nextArrow, {
      [styles.hidden]: selectedImageIndex + 1 === imagesLength,
    });

    const lightboxStyles = classNames(styles.lightbox, { [styles.hidden]: !isOpen });

    const selectedImage = images[selectedImageIndex];
    const pluralImageText = imagesLength !== 1;

    return (
      <div className={lightboxStyles}>
        <div className={styles.background} onClick={onDismiss} />
        <div className={styles.foreground}>
          <div
            className={styles.viewer}
            style={{ backgroundImage: selectedImage ? `url('${selectedImage}')` : 'none' }}
          >
            <div className={styles.overlay}>
              <div className={styles.header}>
                <div className={styles.imageCount}>
                  <div className={styles.imageCountIcon} />
                  <div>
                    {`${selectedImageIndex + 1} `}
                    <FormattedMessage id="lightbox.of">
                      {(message: string) => <span>{message}</span>}
                    </FormattedMessage>
                    {` ${imagesLength} `}
                    <FormattedMessage id={pluralImageText ? 'lightbox.images' : 'lightbox.image'}>
                      {(message: string) => <span>{message}</span>}
                    </FormattedMessage>
                  </div>
                </div>
                <div className={styles.close} onClick={onDismiss} />
              </div>
              <div className={styles.controls}>
                <div className={backArrowStyles} onClick={this.onBackClick} />
                <div className={nextArrowStyles} onClick={this.onNextClick} />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
