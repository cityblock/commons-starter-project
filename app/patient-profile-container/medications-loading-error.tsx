import * as React from 'react';
import * as styles from './css/patient-medications.css';

interface IProps {
  onRetryClick: () => any;
  loading?: boolean;
  error?: string;
}

export const MedicationsLoadingError: React.StatelessComponent<IProps> = props => {
  const { error, loading, onRetryClick } = props;

  if (error) {
    return (
      <div className={styles.medicationsError}>
        <div className={styles.medicationsErrorIcon}></div>
        <div className={styles.medicationsErrorLabel}>Unable to load active medications</div>
        <div className={styles.medicationsErrorSubheading}>
          Sorry, something went wrong. Please try reloading this section again.
        </div>
        <div className={styles.medicationsErrorButton} onClick={onRetryClick}>Try again</div>
      </div>
    );
  } else if (loading) {
    return <div className={styles.medicationsLoading}>Loading...</div>;
  }

  return <div />;
};
