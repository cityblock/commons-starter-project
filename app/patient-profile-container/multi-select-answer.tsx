import * as classNames from 'classnames';
import * as React from 'react';
import { FullAnswerFragment } from '../graphql/types';
import * as styles from './css/risk-areas.css';

interface IProps {
  onClick: (value: string | number, answerId: string) => any;
  answer: FullAnswerFragment;
  editable: boolean;
  selected: boolean;
}

export default class MultiSelectAnswer extends React.Component<IProps, {}> {
  constructor(props: IProps) {
    super(props);

    this.onClick = this.onClick.bind(this);
  }

  componentWillReceiveProps(nextProps: IProps) {
    const { editable } = nextProps;

    if (!editable) {
      this.setState(() => ({ selected: false }));
    }
  }

  onClick() {
    const { editable, answer, onClick } = this.props;

    if (editable) {
      onClick(answer.value, answer.id);
    }
  }

  render() {
    const { answer, editable, selected } = this.props;

    const itemStyles = classNames(styles.multiSelectAnswer, {
      [styles.disabled]: !editable,
      [styles.selected]: selected,
    });

    return (
      <div className={itemStyles} onClick={this.onClick}>
        {answer.displayValue}
      </div>
    );
  }
}
