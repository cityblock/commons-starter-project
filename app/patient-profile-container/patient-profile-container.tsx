import * as classNames from 'classnames';
import * as React from 'react';
import { compose, graphql } from 'react-apollo';
import { injectIntl, FormattedMessage, InjectedIntl } from 'react-intl';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import * as patientQuery from '../graphql/queries/get-patient.graphql';
import { ShortPatientFragment } from '../graphql/types';
import { Size } from '../reducers/browser-reducer';
import * as tabStyles from '../shared/css/tabs.css';
import { IState as IAppState } from '../store';
import * as styles from './css/patient-profile.css';
import PatientCarePlanView from './patient-care-plan-view';
import PatientEncounters from './patient-encounters';
import PatientInfo from './patient-info';
import PatientProfileLeftNav from './patient-profile-left-nav';
import PatientTasks from './patient-tasks';
import PatientThreeSixtyView from './patient-three-sixty-view';

type SelectableTabs = 'encounters' | 'patientInfo' | 'tasks' | '360' | 'carePlan';

interface IProps {
  intl: InjectedIntl;
  patientId: string;
  taskId?: string;
  tabId: SelectableTabs;
  loading: boolean;
  error?: string;
  patient?: ShortPatientFragment;
  browserSize: Size;
  match: {
    params: {
      patientId: string;
      tabId?: SelectableTabs;
      riskAreaOrSubTabId?: string;
    };
  };
}

export const getPatientName = (patient: ShortPatientFragment) => (
  [patient.firstName, patient.middleName, patient.lastName].filter(Boolean).join(' ')
);

class PatientProfileContainer extends React.Component<IProps, {}> {

  componentWillReceiveProps(newProps: IProps) {
    if (newProps.patient) {
      document.title = `${getPatientName(newProps.patient)} | Commons`;
    }
  }

  render() {
    const {
      patientId,
      patient,
      loading,
      error,
      intl,
      tabId,
      taskId,
      browserSize,
      match,
    } = this.props;

    const threeSixtyViewTabStyles = classNames(tabStyles.tab, {
      [tabStyles.selectedTab]: tabId === '360',
    });
    const carePlanTabStyles = classNames(tabStyles.tab, {
      [tabStyles.selectedTab]: tabId === 'carePlan',
    });
    const encountersTabStyles = classNames(tabStyles.tab, {
      [tabStyles.selectedTab]: tabId === 'encounters',
    });
    const patientInfoTabStyles = classNames(tabStyles.tab, {
      [tabStyles.selectedTab]: tabId === 'patientInfo',
    });
    const tasksTabStyles = classNames(tabStyles.tab, {
      [tabStyles.selectedTab]: tabId === 'tasks',
    });

    const mainBodyStyle = classNames({
      [styles.mainBody]: browserSize === 'large',
      [styles.mainBodySmall]: browserSize === 'small',
    });
    const encounters = tabId === 'encounters' ? <PatientEncounters patientId={patientId} /> : null;
    const patientInfo = tabId === 'patientInfo' ?
      <PatientInfo patientId={patientId} patient={patient} loading={loading} error={error} /> :
      null;
    const tasks = tabId === 'tasks' ?
      <PatientTasks patient={patient} taskId={taskId} patientId={patientId} /> :
      null;
    const threeSixty = tabId === '360' ?
      <PatientThreeSixtyView
        riskAreaId={match.params.riskAreaOrSubTabId}
        patientId={patientId}
        routeBase={`/patients/${patientId}/360`} /> : null;
    const carePlan = tabId === 'carePlan' ?
      <PatientCarePlanView
        patientId={patientId}
        routeBase={`/patients/${patientId}/carePlan`}
        subTabId={match.params.riskAreaOrSubTabId as any} /> : null; // TODO: Fix typing
    return (
      <div className={styles.container}>
        <PatientProfileLeftNav
          browserSize={browserSize}
          intl={intl}
          patientId={patientId}
          patient={patient} />
        <div className={mainBodyStyle}>
          <div className={tabStyles.tabs}>
            <FormattedMessage id='patient.threeSixty'>
              {(message: string) =>
                <Link
                  to={`/patients/${patientId}/360`}
                  className={threeSixtyViewTabStyles}>
                  {message}
                </Link>}
            </FormattedMessage>
            <FormattedMessage id='patient.carePlan'>
              {(message: string) =>
                <Link
                  to={`/patients/${patientId}/carePlan`}
                  className={carePlanTabStyles}>
                  {message}
                </Link>}
            </FormattedMessage>
            <FormattedMessage id='patient.encounters'>
              {(message: string) =>
                <Link
                  to={`/patients/${patientId}/encounters`}
                  className={encountersTabStyles}>
                  {message}
                </Link>}
            </FormattedMessage>
            <FormattedMessage id='patient.patientInfo'>
              {(message: string) =>
                <Link
                  to={`/patients/${patientId}/patientInfo`}
                  className={patientInfoTabStyles}>
                  {message}
                </Link>}
            </FormattedMessage>
            <FormattedMessage id='patient.tasks'>
              {(message: string) =>
                <Link
                  to={`/patients/${patientId}/tasks`}
                  className={tasksTabStyles}>
                  {message}
                </Link>}
            </FormattedMessage>
          </div>
          {encounters}
          {patientInfo}
          {tasks}
          {threeSixty}
          {carePlan}
        </div>
      </div>
    );
  }
}

function mapStateToProps(state: IAppState, ownProps: IProps): Partial<IProps> {
  const { browser } = state;
  return {
    browserSize: browser ? browser.size : 'large',
    patientId: ownProps.match.params.patientId,
    tabId: ownProps.match.params.tabId || 'encounters',
  };
}

export default compose(
  injectIntl,
  connect(mapStateToProps),
  graphql(patientQuery as any, {
    options: (props: IProps) => ({
      variables: {
        patientId: props.patientId,
      },
    }),
    props: ({ data }) => ({
      loading: (data ? data.loading : false),
      error: (data ? data.error : null),
      patient: (data ? (data as any).patient : null),
    }),
  }),
)(PatientProfileContainer);
