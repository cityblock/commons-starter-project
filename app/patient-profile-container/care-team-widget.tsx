import * as classNames from 'classnames';
import * as React from 'react';
import { graphql } from 'react-apollo';
import * as careTeamQuery from '../graphql/queries/get-patient-care-team.graphql';
import { FullUserFragment } from '../graphql/types';
import CareTeamWidgetMember from './care-team-widget-member';
import * as careTeamMemberStyles from './css/care-team-widget-member.css';
import * as styles from './css/care-team-widget.css';

interface IProps {
  patientId: string;
  loading?: boolean;
  error?: string;
  careTeam?: FullUserFragment[];
  condensedWidget?: boolean;
}

interface IState {
  open: boolean;
  selectedCareTeamMemberId: string | null;
}

class CareTeamWidget extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);

    const { condensedWidget } = props;

    this.onClick = this.onClick.bind(this);
    this.onCareTeamMemberClick = this.onCareTeamMemberClick.bind(this);
    this.renderCareTeamMember = this.renderCareTeamMember.bind(this);

    this.state = {
      open: condensedWidget ? true : false,
      selectedCareTeamMemberId: null,
    };
  }

  renderCareTeamMember(careTeamMember: FullUserFragment) {
    const selected = careTeamMember.id === this.state.selectedCareTeamMemberId;

    return (
      <CareTeamWidgetMember
        key={careTeamMember.id}
        careTeamMember={careTeamMember}
        onClick={this.onCareTeamMemberClick}
        selected={selected}
      />
    );
  }

  onCareTeamMemberClick(careTeamMemberId: string) {
    this.setState((prevState: IState) => {
      const { selectedCareTeamMemberId } = prevState;

      if (careTeamMemberId === selectedCareTeamMemberId) {
        return { selectedCareTeamMemberId: null };
      } else {
        return { selectedCareTeamMemberId: careTeamMemberId };
      }
    });
  }

  onClick() {
    const { condensedWidget } = this.props;

    if (!condensedWidget) {
      this.setState((prevState: IState) => {
        if (prevState.open) {
          return {
            open: false,
            selectedCareTeamMemberId: null,
          };
        } else {
          return { open: !prevState.open };
        }
      });
    }
  }

  render() {
    const { careTeam, condensedWidget } = this.props;

    const careTeamWidgetClasses = classNames(styles.careTeamWidget, {
      [styles.skinny]: condensedWidget,
    });
    const buttonClasses = classNames(styles.button, {
      [styles.openButton]: this.state.open,
      [styles.condensed]: condensedWidget,
    });
    const drawerClasses = classNames(styles.drawer, {
      [styles.open]: this.state.open,
      [styles.tall]: condensedWidget,
    });
    const buttonIconClasses = classNames(styles.buttonIcon, {
      [styles.hidden]: condensedWidget,
    });
    const buttonArrowClasses = classNames(styles.buttonArrow, {
      [styles.hidden]: condensedWidget,
    });

    const slackContact = (
      <div key='slackContact'>
        <div className={careTeamMemberStyles.careTeamMemberRow}>
          <div className={careTeamMemberStyles.careTeamMemberDetails}>
            <div className={styles.chatLogo} />
            <div className={careTeamMemberStyles.careTeamMemberLabel}>
              <div className={careTeamMemberStyles.careTeamMemberName}>Chat with team</div>
              <div className={careTeamMemberStyles.careTeamMemberRole}>Patient slack channel</div>
            </div>
          </div>
        </div>
        <div className={careTeamMemberStyles.careTeamMemberContact} />
      </div>
    );

    const renderedCareTeamMembers = (careTeam || []).map(this.renderCareTeamMember);

    renderedCareTeamMembers.unshift(slackContact);

    return (
      <div className={careTeamWidgetClasses}>
        <div className={styles.careTeamWidgetRow}>
          <div className={styles.careTeamWidgetContent}>
            <div className={buttonClasses} onClick={this.onClick}>
              <div className={styles.buttonLabel}>
                <div className={buttonIconClasses} />
                <div className={styles.buttonTitle}>Care Team</div>
              </div>
              <div className={buttonArrowClasses} />
            </div>
            <div className={drawerClasses}>{renderedCareTeamMembers}</div>
          </div>
        </div>
      </div>
    );
  }
}

export default graphql(careTeamQuery as any, {
  options: (props: IProps) => ({
    variables: {
      patientId: props.patientId,
    },
  }),
  props: ({ data }) => ({
    loading: data ? data.loading : false,
    error: data ? data.error : null,
    careTeam: data ? (data as any).patientCareTeam : null,
  }),
})(CareTeamWidget);
