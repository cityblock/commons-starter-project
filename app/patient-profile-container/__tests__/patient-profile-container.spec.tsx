import { shallow } from 'enzyme';
import * as React from 'react';
import { PatientProfileContainer as Component, SelectableTabs } from '../patient-profile-container';

const oldDate = Date.now;
const match = {
  params: {
    patientId: 'patient-1',
    tabId: 'map' as SelectableTabs,
  },
};

describe('patient profile container', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 1500494779252);
  });
  afterAll(() => {
    Date.now = oldDate;
  });
  it('renders patient profile container', async () => {
    const component = shallow(
      <Component
        match={match}
        patientId={'patient-1'}
        tabId={'map' as any}
        browserSize={'large'}
        loading={false}
      />,
    );
    const instance = component.instance() as Component;
    const result = instance.render();
    expect(result).toMatchSnapshot();
  });
});
