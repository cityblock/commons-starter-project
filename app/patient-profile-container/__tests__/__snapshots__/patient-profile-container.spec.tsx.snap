// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders patient care plan view 1`] = `
<div>
  <div
    className=""
  >
    <a
      className="undefined"
      href="/patients/patient-1/map/active"
      onClick={[Function]}
    >
      Active
    </a>
    <a
      className=""
      href="/patients/patient-1/map/inactive"
      onClick={[Function]}
    >
      Inactive
    </a>
    <a
      className=""
      href="/patients/patient-1/map/suggestions"
      onClick={[Function]}
    >
      Suggestions
    </a>
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          Loading...
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders patient profile container correctly 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <div>
      <div
        className={undefined}
      />
      <div
        className=""
      >
        <div
          className={undefined}
        >
          <div
            className={undefined}
            style={
              Object {
                "backgroundImage": "url('http://bit.ly/2vaOMQJ')",
              }
            }
          />
          <div
            className={undefined}
          >
            <div
              className={undefined}
            >
              High Risk Patient
            </div>
            <div
              className={undefined}
            >
              <div
                className={undefined}
              >
                Unknown
              </div>
              <div />
            </div>
            <div
              className={undefined}
            >
              40
               • 
            </div>
          </div>
        </div>
        <div
          className={undefined}
        >
          <div
            className={undefined}
          >
            <div>
              Date of birth
              :
            </div>
            <div
              className={undefined}
            >
              2/16/1977
            </div>
          </div>
          <div
            className={undefined}
          >
            <div>
              Preferred language
              :
            </div>
            <div
              className={undefined}
            >
              Declined
            </div>
          </div>
          <div
            className={undefined}
          >
            <div>
              Location
              :
            </div>
            <div
              className={undefined}
            >
              Brooklyn, NY
            </div>
          </div>
          <div
            className={undefined}
          >
            <div>
              Patient since
              :
            </div>
            <div
              className={undefined}
            >
              Unknown
            </div>
          </div>
          <div
            className={undefined}
          >
            <div>
              Medicare ID
              :
            </div>
            <div
              className={undefined}
            >
              123456789
            </div>
          </div>
        </div>
        <div
          className={undefined}
        >
          <div
            className=""
          />
          <div
            className=""
          />
          <div
            className=""
          />
          <div
            className=""
          />
        </div>
        <div
          className={undefined}
        >
          <textarea
            className={undefined}
            disabled={true}
            onChange={[Function]}
            placeholder="Loading..."
            value={undefined}
          />
          <div
            className={undefined}
          />
        </div>
      </div>
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          Active medications
        </div>
        <div
          className={undefined}
        />
      </div>
      <div
        className="undefined"
      >
        <div
          className={undefined}
        >
          Loading...
        </div>
      </div>
    </div>
    <div
      className=""
    >
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <div
            className=""
            onClick={[Function]}
          >
            <div
              className={undefined}
            >
              <div
                className=""
              />
              <div
                className={undefined}
              >
                Care Team
              </div>
            </div>
            <div
              className=""
            />
          </div>
          <div
            className=""
          >
            <div>
              <div
                className={undefined}
              >
                <div
                  className={undefined}
                >
                  <div
                    className={undefined}
                  />
                  <div
                    className={undefined}
                  >
                    <div
                      className={undefined}
                    >
                      Chat with team
                    </div>
                    <div
                      className={undefined}
                    >
                      Patient slack channel
                    </div>
                  </div>
                </div>
              </div>
              <div
                className={undefined}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div
      className={undefined}
    >
      <a
        className=""
        href="/patients/patient-1/360"
        onClick={[Function]}
      >
        360° View
      </a>
      <a
        className="undefined"
        href="/patients/patient-1/map"
        onClick={[Function]}
      >
        MAP
      </a>
      <a
        className=""
        href="/patients/patient-1/timeline"
        onClick={[Function]}
      >
        Timeline
      </a>
      <a
        className=""
        href="/patients/patient-1/patientInfo"
        onClick={[Function]}
      >
        Patient Info
      </a>
    </div>
    <div>
      <div
        className=""
      >
        <a
          className="undefined"
          href="/patients/patient-1/map/active"
          onClick={[Function]}
        >
          Active
        </a>
        <a
          className=""
          href="/patients/patient-1/map/inactive"
          onClick={[Function]}
        >
          Inactive
        </a>
        <a
          className=""
          href="/patients/patient-1/map/suggestions"
          onClick={[Function]}
        >
          Suggestions
        </a>
      </div>
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <div
            className={undefined}
          >
            <div
              className={undefined}
            >
              Loading...
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders patient profile for ipad 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <div
          className={undefined}
        />
        <div
          className={undefined}
          style={
            Object {
              "backgroundImage": "url('http://bit.ly/2vaOMQJ')",
            }
          }
        />
        <div
          className={undefined}
        >
          Loading Patient
        </div>
        <div
          className=""
          onClick={[Function]}
        />
        <div
          className=""
          onClick={[Function]}
        />
      </div>
      <div
        className={undefined}
      >
        <div
          className=""
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="undefined"
    >
      <div>
        <div
          className={undefined}
        />
        <div
          className="undefined"
        >
          <div
            className={undefined}
          >
            <div
              className={undefined}
            >
              <div
                className={undefined}
              >
                High Risk Patient
              </div>
              <div
                className={undefined}
              >
                Unknown null
              </div>
              <div
                className={undefined}
              >
                40
                 • 
              </div>
            </div>
          </div>
          <div
            className={undefined}
          >
            <div
              className={undefined}
            >
              <div>
                Date of birth
                :
              </div>
              <div
                className={undefined}
              >
                2/16/1977
              </div>
            </div>
            <div
              className={undefined}
            >
              <div>
                Preferred language
                :
              </div>
              <div
                className={undefined}
              >
                Declined
              </div>
            </div>
            <div
              className={undefined}
            >
              <div>
                Location
                :
              </div>
              <div
                className={undefined}
              >
                Brooklyn, NY
              </div>
            </div>
            <div
              className={undefined}
            >
              <div>
                Patient since
                :
              </div>
              <div
                className={undefined}
              >
                Unknown
              </div>
            </div>
            <div
              className={undefined}
            >
              <div>
                Medicare ID
                :
              </div>
              <div
                className={undefined}
              >
                123456789
              </div>
            </div>
          </div>
          <div
            className={undefined}
          >
            <div
              className=""
            />
            <div
              className=""
            />
            <div
              className=""
            />
            <div
              className=""
            />
          </div>
          <div
            className={undefined}
          >
            <textarea
              className={undefined}
              disabled={true}
              onChange={[Function]}
              placeholder="Loading..."
              value={undefined}
            />
            <div
              className={undefined}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="undefined"
    >
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <div
            className={undefined}
          >
            Active medications
          </div>
          <div
            className={undefined}
          />
        </div>
        <div
          className="undefined"
        >
          <div
            className={undefined}
          >
            Loading...
          </div>
        </div>
      </div>
    </div>
    <div
      className="undefined"
    >
      <div
        className="undefined"
      >
        <div
          className={undefined}
        >
          <div
            className={undefined}
          >
            <div
              className="undefined"
              onClick={[Function]}
            >
              <div
                className={undefined}
              >
                <div
                  className="undefined"
                />
                <div
                  className={undefined}
                >
                  Care Team
                </div>
              </div>
              <div
                className="undefined"
              />
            </div>
            <div
              className="undefined"
            >
              <div>
                <div
                  className={undefined}
                >
                  <div
                    className={undefined}
                  >
                    <div
                      className={undefined}
                    />
                    <div
                      className={undefined}
                    >
                      <div
                        className={undefined}
                      >
                        Chat with team
                      </div>
                      <div
                        className={undefined}
                      >
                        Patient slack channel
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className={undefined}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className=""
      onClick={[Function]}
    />
  </div>
  <div
    className="undefined"
  >
    <div
      className={undefined}
    >
      <a
        className=""
        href="/patients/patient-1/360"
        onClick={[Function]}
      >
        360° View
      </a>
      <a
        className="undefined"
        href="/patients/patient-1/map"
        onClick={[Function]}
      >
        MAP
      </a>
      <a
        className=""
        href="/patients/patient-1/timeline"
        onClick={[Function]}
      >
        Timeline
      </a>
      <a
        className=""
        href="/patients/patient-1/patientInfo"
        onClick={[Function]}
      >
        Patient Info
      </a>
    </div>
    <div>
      <div
        className=""
      >
        <a
          className="undefined"
          href="/patients/patient-1/map/active"
          onClick={[Function]}
        >
          Active
        </a>
        <a
          className=""
          href="/patients/patient-1/map/inactive"
          onClick={[Function]}
        >
          Inactive
        </a>
        <a
          className=""
          href="/patients/patient-1/map/suggestions"
          onClick={[Function]}
        >
          Suggestions
        </a>
      </div>
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <div
            className={undefined}
          >
            <div
              className={undefined}
            >
              Loading...
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
